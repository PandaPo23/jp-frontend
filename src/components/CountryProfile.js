import React from 'react';
import { inject, observer } from 'mobx-react';
import { withTranslation } from 'react-i18next';
import compose from 'lodash/fp/compose';
import PropTypes from 'prop-types';

import { FlexPanel, ImageSorter, TextContent } from '../design-system';

export const CountryProfile = ({ country, images, mapIsInitialized }) =>
  country ? (
    <FlexPanel bg="background" fontSize={4} flex={1}>
      {mapIsInitialized && (
        <ImageSorter
          height={300}
          minHeight={300}
          borderRadius="none"
          images={images}
        />
      )}
      <TextContent py={0} text={country.description} />
    </FlexPanel>
  ) : null;

CountryProfile.propTypes = {
  country: PropTypes.object,
  t: PropTypes.func.isRequired,
};

export default compose(
  withTranslation('common'),
  inject(({ app }, { countryCode }) => ({
    country: app.countryInfo,
    images: app.getCountryImages(countryCode),
    mapIsInitialized: app.map.initialized,
  })),
  observer
)(CountryProfile);
