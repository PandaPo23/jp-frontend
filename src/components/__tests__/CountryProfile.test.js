import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { CountryProfile } from '../CountryProfile';

jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
  Map: () => ({}),
}));

const country = {
  countryCode: 'JP',
  name: 'Japan',
  description:
    'Japan is distinct - no culture has charmed the world quite like Japanese culture.',
  images: [
    {
      caption: 'Tokyo',
      src:
        'https://jubelproduction.s3.amazonaws.com/uploads/destination/main_image/28/shutterstock_175851101.jpg',
      thumbnail:
        'https://jubelproduction.s3.amazonaws.com/uploads/destination/main_image/28/card_shutterstock_175851101.jpg',
    },
    {
      caption: 'Tokyo',
      src:
        'https://jubelproduction.s3.amazonaws.com/uploads/photo/image/969/shutterstock_268422755.jpg',
      thumbnail:
        'https://jubelproduction.s3.amazonaws.com/uploads/photo/image/969/thumb_shutterstock_268422755.jpg',
    },
  ],
};

describe('CountryProfile component', () => {
  it('renders correctly', () => {
    const localProps = {
      country,
      t: (k, v) => v,
    };
    const wrapper = shallow(<CountryProfile {...localProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
