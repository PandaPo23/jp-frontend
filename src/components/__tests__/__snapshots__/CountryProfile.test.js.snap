// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryProfile component renders correctly 1`] = `
<FlexPanel
  bg="background"
  display="flex"
  flex={1}
  flexDirection="column"
  fontSize={4}
>
  <TextContent
    py={0}
    text="Japan is distinct - no culture has charmed the world quite like Japanese culture."
  />
</FlexPanel>
`;
